rules/CodeQuality/Rector/ClassMethod/InlineArrayReturnAssignRector.php

```diff
@@ -187,6 +188,12 @@ private function areAssignExclusiveToDimFetch(array $stmts): bool
             }
 
             $assign = $stmt->expr;
+
+            // skip new X instance with args to keep complex assign readable
+            if ($assign->expr instanceof New_ && $assign->expr->getArgs() !== []) {
```

Comment:
`isFirstClassCallable()` need to be checked, or just use direct `count($assign->expr->args) > 0` since `getArgs()` use assert

Reply:
I created PR :

- https://github.com/rectorphp/rector-src/pull/6763

to handle that.